src/modules/keycloak-administration/repository/user-lock.repository.integration-spec.ts

```diff
@@ -52,39 +53,61 @@ describe('UserLockRepository', () => {
     });
 
     describe('findPersonById', () => {
-        it('should return UserLock when found by person', async () => {
+        it('should return an array of UserLocks when found by person', async () => {
             const newPerson: PersonEntity = createPersonEntity();
             await em.persistAndFlush(newPerson);
-            const userLock: UserLock = UserLock.construct(newPerson.id, faker.string.uuid(), new Date(), new Date());
 
-            const createdUserLock: UserLock | DomainError = await sut.createUserLock(userLock);
-            if (createdUserLock instanceof DomainError) throw new Error();
-            expect(createdUserLock).toBeTruthy();
+            // Create multiple UserLocks for the same person to test array retrieval
+            const userLock1: UserLock = UserLock.construct(
+                newPerson.id,
+                faker.string.uuid(),
+                new Date(),
+                PersonLockOccasion.MANUELL_GESPERRT,
+                new Date(),
+            );
+            const userLock2: UserLock = UserLock.construct(
+                newPerson.id,
+                faker.string.uuid(),
+                new Date(),
+                PersonLockOccasion.KOPERS_GESPERRT,
+                new Date(),
+            );
+
+            await sut.createUserLock(userLock1);
+            await sut.createUserLock(userLock2);
 
-            const foundUserLock: Option<UserLock> = await sut.findPersonById(userLock.person);
+            const foundUserLocks: Option<UserLock[]> = await sut.findByPersonId(userLock1.person);
```

Comment:
A lot of types are now "incorrect" in this file. Should not be `Option<UserLock[]>`